goagen/gen_client/generator.go

```diff
@@ -1011,6 +1014,13 @@ func (c *Client) {{ $funcName }}(ctx context.Context, path string{{ if .Params }
 	if err != nil {
 		return nil, err
 	}
+	if c.BasicAuthSigner != nil {
+		wsSigner, ok := c.BasicAuthSigner.(*goaclient.BasicSigner)
```

Comment:
It would be better to have explicit fields on the client struct for each possible web socket signer type. So something like:
```go
if c.BasicAuthWSSigner != nil {
    c.BasicAuthWSSigner.Sign(cfg)
}
```
Also the code should check for the proper security scheme and use the corresponding field (`BasicAuthWSSigner` or `APIKeyWSSigner` or ...) like is done in the existing signer case.